picky/src/x509/pkcs7/authenticode.rs

```diff
@@ -1159,21 +1191,22 @@ impl<'a> AuthenticodeSignatureBuilder<'a> {
             .take()
             .ok_or(AuthenticodeSignatureBuilderError::MissingCertificatesRequired)?;
 
+        let signing_certificate = certificates
+            .iter()
+            .find(|cert| {
+                Name::from(cert.issuer_name()) == issuer_and_serial_number.issuer
+                    && cert.serial_number() == &issuer_and_serial_number.serial_number.0
+            })
+            .ok_or(AuthenticodeError::NoCertificatesAssociatedWithIssuerAndSerialNumber {
+                issuer: issuer_and_serial_number.issuer.clone(),
+                serial_number: issuer_and_serial_number.serial_number.0 .0.clone(),
+            })
```

Comment:
This should probably be `ok_or_else` because of the `clone`s (which are deep copies in this case). You may try to run `cargo clippy --all-features` to make sure clippy is checking this module.

(I'm think of [this lint](https://rust-lang.github.io/rust-clippy/master/#or_fun_call))